src/net_processing.cpp

```diff
@@ -2365,7 +2365,13 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         m_connman.PushMessage(&pfrom, msg_maker.Make(NetMsgType::VERACK));
 
         // Signal ADDRv2 support (BIP155).
-        m_connman.PushMessage(&pfrom, msg_maker.Make(NetMsgType::SENDADDRV2));
+        if (greatest_common_version >= 70016) {
+            // BIP155 defines addrv2 and sendaddrv2 for all protocol versions, but some
+            // implementations reject messages they don't know. As a courtesy, don't send
+            // it to nodes with a version before 70016, as no software is known to support
+            // BIP155 that doesn't announce at least that protocol version number.
```

Comment:
nit: Could mention that this is a temporary workaround, safe to be reverted end of 2021 or beginning of 2022

Reply:
We're not prophets...

Reply:
This change literally violates the bip, keeping the hack longer than one or two major versions would be problematic for too many reasons

Reply:
I don't see how it violates the BIP. It just chooses to only use the feature in some cases.

Reply:
Ok, closing discussion. I still think this should be reverted eventually.

Reply:
Yes, agree with that, but I don't know the time frame.